azure/activity_logs_monitoring/index.js

```diff
@@ -77,6 +81,164 @@ class ScrubberRule {
     }
 }
 
+class Batcher {
+    constructor(context, maxItemSizeBytes, maxBatchSizeBytes, maxItemsCount) {
+        this.maxItemSizeBytes = maxItemSizeBytes;
+        this.maxBatchSizeBytes = maxBatchSizeBytes;
+        this.maxItemsCount = maxItemsCount;
+    }
+
+    batch(items) {
+        var batches = [];
+        var batch = [];
+        var sizeBytes = 0;
+        var sizeCount = 0;
+        items.forEach(item => {
+            var itemSizeBytes = this.getSizeInBytes(item);
+            if (
+                sizeCount > 0 &&
+                (sizeCount >= this.maxItemsCount ||
+                    sizeBytes + itemSizeBytes > this.maxBatchSizeBytes)
+            ) {
+                batches.push(batch);
+                batch = [];
+                sizeBytes = 0;
+                sizeCount = 0;
+            }
+            // all items exceeding maxItemSizeBytes are dropped here
+            if (itemSizeBytes <= this.maxItemSizeBytes) {
+                batch.push(item);
+                sizeBytes += itemSizeBytes;
+                sizeCount += 1;
+            }
+        });
+
+        if (sizeCount > 0) {
+            batches.push(batch);
+        }
+        return batches;
+    }
+
+    getSizeInBytes(string) {
+        if (typeof string !== 'string') {
+            string = JSON.stringify(string);
+        }
+        return Buffer.byteLength(string, 'utf8');
+    }
+}
+
+class HTTPClient {
+    constructor(context) {
+        this.context = context;
+        this.httpOptions = {
+            hostname: DD_HTTP_URL,
+            port: DD_HTTP_PORT,
+            path: '/v1/input',
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'DD-API-KEY': DD_API_KEY
+            }
+        };
+        this.scrubber = new Scrubber(this.context, SCRUBBER_RULE_CONFIGS);
+        this.batcher = new Batcher(
+            this.context,
+            256 * 1000,
+            4 * 1000 * 1000,
+            400
+        );
+    }
+
+    async sendAll(records) {
+        var batches = this.batcher.batch(records);
+        var promises = [];
+        for (var i = 0; i < batches.length; i++) {
```

Comment:
*nit: (extremely minor) seeing a mix of forEach and traditional fors for iterating over lists.